### PR TITLE
[TM-1237] Image Gallery Enhancement

### PR DESCRIPTION
**Pull request by dottyy**
[Task Link](https://gfw.atlassian.net/jira/software/c/projects/TM/boards/40?selectedIssue=TM-1237)

### CODE REVIEW

File: src/components/elements/ImageGallery/ImageGallery.tsx [hunk 1]
```diff
@@ -13,6 +13,9 @@ import { VARIANT_PAGINATION_TEXT_16 } from "@/components/extensive/Pagination/Pa
 import { useModalContext } from "@/context/modal.provider";
 import { Option } from "@/types/common";
 
+import Button from "../Button/Button";
+import FilterSearchBox from "../TableFilters/Inputs/FilterSearchBox";
+import Toggle from "../Toggle/Toggle";
```
Comment (pachonjcl) [inline src/components/elements/ImageGallery/ImageGallery.tsx:18]: please change this imports to not use relative paths `../Button...` should be `@/...`

File: src/components/elements/ImageGallery/__snapshots__/ImageGalleryItem.stories.storyshot [hunk 2]
```diff
@@ -55,70 +69,126 @@ exports[`Storyshots Components/Elements/Cards/ImageGalleryCard/Item Default 1`]
...
-              className="flex items-center justify-center rounded-full hover:bg-opacity-60 z-10 hover:opacity-60"
-              onClick={[Function]}
-            >
               <div
+                className="absolute z-40 top-full mt-[5px] invisible"
                 style={
                   Object {
-                    "height": 32,
-                    "width": 32,
+                    "width": "undefinedpx",
```
Comment (pachonjcl) [inline src/components/elements/ImageGallery/__snapshots__/ImageGalleryItem.stories.storyshot:88]: can you fix this story shot?

File: src/components/elements/Menu/Menu.tsx [hunk 3]
```diff
@@ -187,35 +188,41 @@ const Menu = (props: MenuProps) => {
...
-                if (item?.onClick) {
-                  if (item?.is_airtable) {
-                    item?.onClick(extraData);
-                  } else {
-                    item?.onClick();
+            <>
+              {item?.type === "line" ? (
+                <div className="mx-2 border-b border-[#d7dbdc]" />
+              ) : (
+                <MenuItem
```
Comment (pachonjcl) [inline src/components/elements/Menu/Menu.tsx:195]: Please use the `When` component from react-if

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/ImageGallery/ImageGallery.tsx (hunk 1 above)
Comment (dottyy) [reply src/components/elements/ImageGallery/ImageGallery.tsx]: done

File: src/components/elements/Menu/Menu.tsx (hunk 3 above)
Comment (dottyy) [reply src/components/elements/Menu/Menu.tsx]: done

File: src/components/elements/ImageGallery/__snapshots__/ImageGalleryItem.stories.storyshot (hunk 2 above)
Comment (dottyy) [reply src/components/elements/ImageGallery/__snapshots__/ImageGalleryItem.stories.storyshot]: done

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**